x-pack/plugins/apm/public/components/shared/unified_search_bar/index.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  Filter,
+  fromKueryExpression,
+  Query,
+  TimeRange,
+  toElasticsearchQuery,
+} from '@kbn/es-query';
+import { useHistory, useLocation } from 'react-router-dom';
+import deepEqual from 'fast-deep-equal';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { EuiSkeletonRectangle } from '@elastic/eui';
+import qs from 'query-string';
+import { DataView, UI_SETTINGS } from '@kbn/data-plugin/common';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { UIProcessorEvent } from '../../../../common/processor_event';
+import { TimePickerTimeDefaults } from '../date_picker/typings';
+import { ApmPluginStartDeps } from '../../../plugin';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmDataView } from '../../../hooks/use_apm_data_view';
+import { useProcessorEvent } from '../../../hooks/use_processor_event';
+import { fromQuery, toQuery } from '../links/url_helpers';
+import { useApmParams } from '../../../hooks/use_apm_params';
+import { getBoolFilter } from '../get_bool_filter';
+import { useLegacyUrlParams } from '../../../context/url_params_context/use_url_params';
+
+function useSearchBarParams(defaultKuery?: string) {
+  const { path, query } = useApmParams('/*');
+  const kuery = 'kuery' in query ? query.kuery : undefined;
+  const serviceName = 'serviceName' in path ? path.serviceName : undefined;
+  const groupId = 'groupId' in path ? path.groupId : undefined;
+  const environment = 'environment' in query ? query.environment : undefined;
+
+  return {
+    urlQuery: kuery
```

Comment:
[nit] shouldn't this be just `kuery`? we are either setting the `urlKuery` or the `defaultKuery` here?